From: Trond Myklebust <Trond.Myklebust@netapp.com>
Date: Tue, 24 Sep 2013 12:06:07 -0400
Subject: NFSv4: Ensure that we disable the resend timeout for NFSv4
Git-commit: 99875249bfbfb6d9a2aba020ce65da2862d0dafa
Patch-mainline: v3.13
References: bnc#868923

The spec states that the client should not resend requests because
the server will disconnect if it needs to drop an RPC request.

Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
Acked-by: NeilBrown <neilb@suse.de>

---
 fs/nfs/client.c           |    2 ++
 include/linux/nfs_fs_sb.h |    1 +
 2 files changed, 3 insertions(+)

--- linux-3.12-SLE12.orig/fs/nfs/client.c
+++ linux-3.12-SLE12/fs/nfs/client.c
@@ -598,6 +598,8 @@ int nfs_create_rpc_client(struct nfs_cli
 
 	if (test_bit(NFS_CS_DISCRTRY, &clp->cl_flags))
 		args.flags |= RPC_CLNT_CREATE_DISCRTRY;
+	if (test_bit(NFS_CS_NO_RETRANS_TIMEOUT, &clp->cl_flags))
+		args.flags |= RPC_CLNT_CREATE_NO_RETRANS_TIMEOUT;
 	if (test_bit(NFS_CS_NORESVPORT, &clp->cl_flags))
 		args.flags |= RPC_CLNT_CREATE_NONPRIVPORT;
 	if (test_bit(NFS_CS_INFINITE_SLOTS, &clp->cl_flags))
--- linux-3.12-SLE12.orig/include/linux/nfs_fs_sb.h
+++ linux-3.12-SLE12/include/linux/nfs_fs_sb.h
@@ -42,6 +42,7 @@ struct nfs_client {
 #define NFS_CS_MIGRATION	2		/* - transparent state migr */
 #define NFS_CS_INFINITE_SLOTS	3		/* - don't limit TCP slots */
 #define NFS_CS_NO_SHARE		4		/* - don't share across mounts */
+#define NFS_CS_NO_RETRANS_TIMEOUT	5	/* - Disable retransmit timeouts */
 	struct sockaddr_storage	cl_addr;	/* server identifier */
 	size_t			cl_addrlen;
 	char *			cl_hostname;	/* hostname of server */
